src/utils.py

```diff
@@ -252,6 +252,13 @@ def push_metrics_to_pushgateway(metric, metric_value, label_value, registry):
         logger.error(f"Failed to push metrics: {e}")
 
 
+def dict_in_list(target_dict, dict_list):
+    for d in dict_list:
+        if d == target_dict:
+            return True
+    return False
```

Comment:
sorry, my python is a bit rusty, but isn't this equivalent to

```suggestion
    return target_dict in dict_list
```

Reply:
should be the same 👍